File: 807-custom-sort-string/custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        count = Counter(s)
        ans = []
        for ch in order:
            if ch in count:
                ans.append(ch*count[ch])
                del count[ch]
        for ch in count:
            ans.append(ch*count[ch])
        return "".join(ans)

File: 807-custom-sort-string/custom-sort-string.cs
public class Solution {
    public string CustomSortString(string order, string s) {
        Dictionary<char,int> count = new();
        foreach(var ch in s) {
            if (!count.ContainsKey(ch)) {
                count[ch] = 0;
            }
            count[ch] += 1;
        }

        List<char> ans = new();
        foreach (var ch in order) {
            if (count.ContainsKey(ch)) {
                for (int i=0;i<count[ch];i++)
                    ans.Add(ch);
                count.Remove(ch);
            }
        }

        foreach (var ch in count.Keys) {
            for (int i=0;i<count[ch];i++) {
                ans.Add(ch);
            }
        }

        return new string(ans.ToArray());
    }
}